packages/components/input/src/input.tsx

```diff
@@ -1,29 +1,167 @@
 import { input, InputStylesProps } from '@consolelabs/theme'
-import { HTMLAttributes } from 'react'
+import React, { InputHTMLAttributes, useMemo } from 'react'
+import type * as Polymorphic from '@consolelabs/polymorphic'
 
-type InputProps = HTMLAttributes<HTMLInputElement> &
-  InputStylesProps & {
-    disabled?: boolean
-    error?: boolean
-    className?: string
-  }
+// root
+
+type TextFieldContextValue = {
+  size?: 'md' | 'lg'
+  disabled?: boolean
+  error?: boolean
+  spacing?: 'md' | 'lg'
```

Comment:
I'm not sure about the design but ideally the `size` property is enough, we don't need the `spacing` property.